### PR TITLE
Reduce CI load

### PR DESCRIPTION
**Pull request by DanNixon**
To test:

- Review changes to CI config
- See that formatting always runs (it should have ran on this PR)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,23 +41,25 @@ jobs:
...
       - name: Clippy
+        if: steps.filter.outputs.rust == 'true'
         run: nix develop --command cargo clippy --all-targets -- -D warnings
 
       - name: Repository cleanliness
+        if: steps.filter.outputs.rust == 'true'
         run: git diff --exit-code
 
       - name: Test
+        if: steps.filter.outputs.flake == 'true' && steps.filter.outputs.rust == 'true'
```
Comment (Modularius) [inline .github/workflows/ci.yml:52]: Wouldn't this only run if both rust and nix files have been changed? Maybe it should be
```nix
if: steps.filter.outputs.flake == 'true' || steps.filter.outputs.rust == 'true'
```

Comment (DanNixon) [reply .github/workflows/ci.yml]: Ah, yes. Good catch.

File: .github/workflows/component.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+              - 'Cargo.lock'
+              # Rust common
+              - 'common/Cargo.toml'
+              - 'common/**/*.rs'
+              - 'streaming-types/Cargo.toml'
+              - 'streaming-types/**/*.rs'
+              # Component
+              - "${{ inputs.component }}/flake.nix"
+              - "${{ inputs.component }}/Cargo.toml"
+              - "${{ inputs.component }}/**/*.rs"
```
Comment (Modularius) [inline .github/workflows/component.yml:38]: Should there also be filters for the default.nix files and README.md in each component?

Comment (DanNixon) [reply .github/workflows/component.yml:38]: Ah, that was supposed to be `default.nix` not `flake.nix` there.

The readme is not required as this has no influence over the binary.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -28,23 +41,25 @@ jobs:
         run: nix develop --command treefmt --fail-on-change
 
       - name: Clippy
+        if: steps.filter.outputs.rust == 'true'
         run: nix develop --command cargo clippy --all-targets -- -D warnings
 
       - name: Repository cleanliness
+        if: steps.filter.outputs.rust == 'true'
```
Comment (Modularius) [inline .github/workflows/ci.yml:48]: Also, I think this step would be needed if nix files had been edited too.

**Review by Modularius — APPROVED** [review]

**Pull request merged**